hdf/src/df24f.c

```diff
@@ -104,7 +104,7 @@ nd2igdim(_fcd filename, intf *pxdim, intf *pydim, intf *pil, intf *fnlen)
     if (!fn)
         return (-1);
     ret = DF24getdims(fn, (int32 *)pxdim, (int32 *)pydim, (intn *)pil);
-    HDfree((VOIDP)fn);
+    free(fn);
```

Comment:
I don't know where anyone got the idea that you had to cast a pointer being freed to void * or char *, but man there were a lot of instances of this.

Reply:
Probably some arcane compiler warning!

Reply:
I think both HDF4 and HDF5 contain a lot of unnecessary pedantry that clouds reading.